static/app/views/settings/organizationTeams/teamMembers.tsx

```diff
@@ -167,36 +161,28 @@ function AddMemberDropdown({
                 memberId: selection.value,
               })
       }
-      menuHeader={
-        <StyledMembersLabel>
-          {t('Members')}
-          <StyledCreateMemberLink
-            to=""
-            onClick={() => openInviteMembersModal({source: 'teams'})}
-            data-test-id="invite-member"
-          >
-            {t('Invite Member')}
-          </StyledCreateMemberLink>
-        </StyledMembersLabel>
+      menuHeaderTrailingItems={
+        <StyledCreateMemberLink
+          to=""
+          onClick={() => openInviteMembersModal({source: 'teams'})}
+          data-test-id="invite-member"
+        >
+          {t('Invite Member')}
+        </StyledCreateMemberLink>
       }
-      emptyMessage={t('No members')}
-      onChange={handleMemberFilterChange}
-      onClose={() => setMemberQuery('')}
       disabled={isDropdownDisabled}
-      data-test-id="add-member-menu"
-      busy={isOrgMembersLoading}
-    >
-      {({isOpen}) => (
-        <DropdownButton
-          isOpen={isOpen}
-          size="xs"
-          data-test-id="add-member"
-          disabled={isDropdownDisabled}
-        >
-          {t('Add Member')}
-        </DropdownButton>
-      )}
-    </DropdownAutoComplete>
+      menuTitle={t('Members')}
+      triggerLabel={t('Add Member')}
+      searchPlaceholder={t('Search Members')}
+      emptyMessage={t('No members')}
+      loading={isOrgMembersFetching}
+      searchable
+      disableSearchFilter
+      /**
+       * We perform an API request to support orgs with > 100 members (since that's the max API returns)
+       */
+      onSearch={setMemberQuery}
```

Comment:
note: the API around `search` is pretty bad here. `searchable` adds the search field, but then we need to `disableSearchFilter` to stop it from doing filtering, so that `onSearch` can be applied here to do the filtering in a controlled way - yet we still don’t pass the `search` value we have stored here for filtering to it. It’s pretty messy.

I don’t think we need `onSearch` when we do an uncontrolled search, so one approach would be:

```
searchable: true | { searchValue: string, onSearch: (value: string) => void
```

this would rid us of `disableSearchFilter`, and it would make sure you’d have to pass `searchValue` alongside `onSearch` as both are required.

Reply:
Mind leaving a note with a link to this comment in our component audit page? This is gross, and very unintuitive